src/components/Link/Link.tsx

```diff
@@ -22,29 +25,54 @@ export interface BaseProps {
 }
 
 export interface Props extends BaseProps {}
+export type CombinedProps = Props & WithAppProviderProps;
 
-export default function Link({
+function Link({
   url,
   children,
   onClick,
   external,
   id,
   monochrome,
-}: Props) {
+  polaris,
+}: CombinedProps) {
   const className = classNames(styles.Link, monochrome && styles.monochrome);
+  let childrenMarkup = children;
+
+  if (external && typeof children === 'string') {
+    const lastWord = children.slice(children.lastIndexOf(' ') + 1);
+    const rest = children.slice(0, children.lastIndexOf(' ') + 1);
+    const iconLabel = polaris.intl.translate(
+      'Polaris.Common.newWindowAccessibilityHint',
+    );
+
+    childrenMarkup = (
+      <React.Fragment>
+        {rest}
+        <span className={styles.IconLockup}>
+          {lastWord}
+          <Icon accessibilityLabel={iconLabel} source={ExternalSmallMinor} />
+        </span>
+      </React.Fragment>
+    );
+  }
+
```

Comment:
`Link` in Polaris was originally intended only for text links within sentences. In practice, `Link` is used more like React Router’s `Link`: a generic navigation routing component. As such, it can be used to wrap images and block-level elements, and is typed accordingly (`children` accepts React nodes). This makes determining the last word in `children` impossible where `children` is not a string.

This PR is solving only for the case where Link is used as a text link, and is passed a single string in `children`. Based on a quick audit of real-world usage, this is the common case.